#!/usr/bin/env python
PACKAGE = "coordinated_motion_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("alpha", double_t, 0, "The damping value for the Jacobian pseudoinverse", 0.1, 0.0, 1.0)
gen.add("k_position", double_t, 0, "The gain to scale position error", 1.0, 0.0, 100)
gen.add("k_aiming", double_t, 0, "The gain to scale aiming error", 1.0, 0.0, 100)
gen.add("k_manip", double_t, 0, "The gain to scale the manipulability maximization", 1.0, 0.0, 500)
gen.add("k_limits", double_t, 0, "The gain to scale the joint limit avoidance", 1.0, 0.0, 500)

exit(gen.generate(PACKAGE, "coordinated_motion_controllers", "CoordinatedController"))
